tests/physics/helicity_formalism/test_amplitude.py

```diff
@@ -77,7 +78,11 @@ def test_coherent_intensity(functions, test_data, expected_results):
 def test_coefficient_amplitude(
     function, mag, phase, test_data, expected_results
 ):
-    model = _CoefficientAmplitude(function, mag, phase)
+    model = _CoefficientAmplitude(
+        function,
+        tf.constant(mag, dtype=tf.float64),
```

Comment:
Do we leave the responsibility of chooseing the `dtype` to the caller?

Reply:
With the introduction of the prefactor, the `_CoefficientAmplitude` expect, the magnitude and phase to be `float64`. Maybe the datatype can also be inferred from the magnitude and phase and use the same type for the prefactor to achieve higher compatiblity. For the amplitudes generated by the builder, everything is `float64` though.